server/backend.go

```diff
@@ -32,7 +36,13 @@ type Backend struct {
 }
 
 func NewBackend(id string, addr string) (*Backend, error) {
-	client, err := grpc.Dial(addr, grpc.WithInsecure())
+	creds := insecure.NewCredentials()
+	if val := os.Getenv("TLS_SKIP_VERIFY"); val != "" {
+		creds = credentials.NewTLS(&tls.Config{
```

Comment:
tls is required when the index-reading-backend is setup as an independent (cloud run) service on gcs.